modules/ting_das/ting_das.module

```diff
@@ -0,0 +1,378 @@
+    'delivery callback' => 'ajax_deliver',
+    'access arguments' => array('access content'),
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_libraries_info().
+ *
+ * For defining external libraries.
+ */
+function ting_das_libraries_info() {
+  return array(
+    'guzzle' => array(
+      'name' => 'Guzzle',
+      'vendor url' => 'https://github.com/guzzle/guzzle',
+      'download url' => 'https://github.com/guzzle/guzzle',
+      'version' => '6.2',
+      'xautoload' => function ($adapter) {
+        $adapter->composerJson('composer.json');
+      },
+    ),
+  );
+}
+
+/**
+ * Implements hook_ding_entity_buttons().
+ */
+function ting_das_ding_entity_buttons($type, $entity, $view_mode, $widget = 'default') {
+  $type = drupal_strtolower($entity->getType());
+  $types = array_filter(_ting_das_get_default_well_types());
+
+  // Check that the entity is not already available as an online resources and is one of the configured types.
+  if (isset($types[$type]) && !$entity->is('online') && !$entity->is('infomedia')) {
+    drupal_add_library('system', 'drupal.ajax');
+
+    $display_button = TRUE;
+    if (user_is_logged_in()) {
+      try {
+        $creds = ding_user_get_creds();
+        $display_button = isset($creds['resident']) && $creds['resident'];
+      }
+      catch (DingProviderAuthException $exception) {
+        // This is okay as the use simply was not logged in as an provider
+        // user. So do noting in this case.
+      }
+    }
+
+    if ($display_button) {
+      return array(
+        array(
+          '#theme' => 'link',
+          '#text' => t('Order digital copy'),
+          '#path' => 'das/' . $entity->id . '/order',
+          '#options' => array(
+            'attributes' => array(
+              'class' => array(
+                'action-button',
+                'button-see-online',
+                'use-ajax',
+              ),
+              'id' => 'das-order-' . $entity->id,
+              'onclick' => 'wt.sendinfo({ linkId: "open_order_digital_copy" });',
+            ),
+            'html' => FALSE,
+          ),
+          '#attached' => array(
+            'css' => array(
+              drupal_get_path('module', 'ting_das') . '/css/ting_das.css',
+            ),
+          ),
+        ),
+      );
+    }
+  }
+}
+
+/**
+ * Entity button ajax callback handler.
+ *
+ * @param TingEntity $entity
+ *   The entity that the order is about.
+ *
+ * @return array
+ *   Ajax commands based on the current state and the entity given.
+ */
+function ting_das_ajax($entity) {
+  $commands = array();
+
+  // Check if the logged in user is a library user.
+  global $user;
+  if (!user_is_logged_in()) {
+    $commands[] = ajax_command_ding_user_authenticate('');
+  }
+  elseif (!ding_user_is_provider_user($user)) {
+    // Error not library user.
+    $commands[] = ajax_command_ding_popup('ting_das', t('Error'), '<p>' . t('Only library user can make DAS requests.') . '</p>', array('refresh' => FALSE));
```

Comment:
I doubt end users will know what a "DAS request" is.

Reply:
Changes, but it's only to users that is logged in as non-library users.